wallet/udb/upgrades.go

```diff
@@ -684,6 +691,49 @@ func hasExpiryFixedUpgrade(tx walletdb.ReadWriteTx, publicPassphrase []byte) err
 	return unifiedDBMetadata{}.putVersion(metadataBucket, newVersion)
 }
 
+func removeExtraMultisigOutsUpgrade(tx walletdb.ReadWriteTx, publicPassphrase []byte) error {
+	const oldVersion = 9
+	const newVersion = 10
+	metadataBucket := tx.ReadWriteBucket(unifiedDBMetadata{}.rootBucketKey())
+	txmgrBucket := tx.ReadWriteBucket(wtxmgrBucketKey)
+
+	// Assert this function is only called on version 9 databases.
+	dbVersion, err := unifiedDBMetadata{}.getVersion(metadataBucket)
+	if err != nil {
+		return err
+	}
+	if dbVersion != oldVersion {
+		const str = "removeExtraMultisigOutsUpgrade inappropriately called"
+		return errors.E(errors.Invalid, str, nil)
+	}
+
+	var multisigOutsMarkedForDelete [][]byte
+	multisigOutBucket := txmgrBucket.NestedReadWriteBucket([]byte("ms"))
+
+	c := multisigOutBucket.ReadCursor()
+	for k, v := c.First(); k != nil; k, v = c.Next() {
+		mso, err := fetchMultisigOut(k, v)
+		if err != nil {
+			str := "failed to fetch multisig from bucket"
+			return errors.E(errors.IO, str, nil)
+		}
+
+		_, txVal := existsTxRecord(txmgrBucket, &mso.TxHash, &Block{
+			Hash:   mso.BlockHash,
+			Height: int32(mso.BlockHeight),
+		})
+		if txVal == nil {
+			multisigOutsMarkedForDelete = append(multisigOutsMarkedForDelete, k)
```

Comment:
to be on the safe side, make a copy of the key.  bolt is unfortunately ridden with bugs related to cursor invalidation

Reply:
@jrick could you check this change in fresh version. Did I make it right?